packages/flutter/test/foundation/profile_test.dart

```diff
@@ -0,0 +1,18 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:test/test.dart';
+
+const bool isReleaseMode = const bool.fromEnvironment("dart.vm.product");
+
+void main() {
+  test("profile invokes its closure in debug or profile mode", () {
+    int count = 0;
+    profile(() {
+      count++;
+    });
+    expect(count, isReleaseMode ? 0 : 1);
+  });
+}
```

Comment:
There's really no good way for us to test that `profile` actually omits the code in release builds in the current infrastructure. At some point we should probably do a real test for this.